smbclient/_os.py

```diff
@@ -75,10 +75,11 @@
     DateTimeField,
 )
 
-
 XATTR_CREATE = getattr(os, 'XATTR_CREATE', 1)
 XATTR_REPLACE = getattr(os, 'XATTR_REPLACE', 2)
 
+MAX_COPY_CHUNK_SIZE = 1 * 1024 * 1024  # maximum chunksize 1M from 3.3.3 in MS-SMB documentation
```

Comment:
I'm still not sure where this size comes from? MS-SMB is the SMBv1 implementation which is a different protocol from SMBv2 (MS-SMB2). I also can't find mention of this in the docs on MS-SMB2 talking about receiving this request at https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb2/cd0162e4-7650-4293-8a2a-d696923203ef.

Maybe it's worth trying out a copy of a large file to see when it starts to fail to verify everything.

Reply:
Ok I did some testing myself and can confirm those limits are still in place for SMB2. You can actually see the limit documented on the raw IOCTL docs for this command at https://docs.microsoft.com/en-us/windows/win32/devnotes/ioctl-copychunk.

The `Length` field states

> The number of bytes of data in the chunk to be copied. Must be greater than zero and less than or equal to 1 MB. Length * ChunkCount must be less than or equal to 16 MB.

So this confirms the max size of `1024 * 1024` for a chunk size and 16 chunks per request. Sorry for doubting you here :)

Reply:
No worries. I have extracted it as constants and have added a proper description.